packages/core/test/VDocument.test.ts

```diff
@@ -1223,5 +1228,644 @@ describe('stores', () => {
+                                contentAfter: '<p><br></p><p>[]abc</p>',
+                            });
+                            testEditor({
+                                contentBefore: '<p>[] abc</p>',
+                                stepFunction: enter,
+                                // The space should have been parsed away.
+                                contentAfter: '<p><br></p><p>[]abc</p>',
+                            });
+                        });
+                        it('should split a paragraph in two', () => {
+                            testEditor({
+                                contentBefore: '<p>ab[]cd</p>',
+                                stepFunction: enter,
+                                contentAfter: '<p>ab</p><p>[]cd</p>',
+                            });
+                            testEditor({
+                                contentBefore: '<p>ab []cd</p>',
+                                stepFunction: enter,
+                                // The space is converted to a non-breaking
+                                // space so it is visible.
+                                contentAfter: '<p>ab&nbsp;</p><p>[]cd</p>',
+                            });
+                            testEditor({
+                                contentBefore: '<p>ab[] cd</p>',
+                                stepFunction: enter,
+                                // The space is converted to a non-breaking
+                                // space so it is visible.
+                                contentAfter: '<p>ab</p><p>[]&nbsp;cd</p>',
+                            });
+                        });
+                        it('should insert an empty paragraph after a paragraph', () => {
+                            testEditor({
+                                contentBefore: '<p>abc[]</p>',
+                                stepFunction: enter,
+                                contentAfter: '<p>abc</p><p>[]<br></p>',
+                            });
+                            testEditor({
+                                contentBefore: '<p>abc []</p>',
+                                stepFunction: enter,
+                                // The space should have been parsed away.
+                                contentAfter: '<p>abc</p><p>[]<br></p>',
+                            });
+                        });
+                    });
+                    describe('Consecutive', () => {
+                        it('should duplicate an empty paragraph twice', () => {
+                            testEditor({
+                                contentBefore: '<p>[]<br></p>',
+                                stepFunction: (editor: JWEditor) => {
+                                    enter(editor);
+                                    enter(editor);
+                                },
+                                contentAfter: '<p><br></p><p><br></p><p>[]<br></p>',
```

Comment:
Not absolutely necessary indeed. Kind of just an extra security for my peace of mind, in case we had some kind of false positive before : this effectively tests that the range was set within the first duplicated empty P in the same way as at the beginning of the test, or that whatever it was set to, it did its job right. More than two times is definitely not needed ;-)